lib/pdk/module/convert.rb

```diff
@@ -76,6 +72,56 @@ def self.update_metadata(metadata_path, template_metadata)
         metadata.update!(template_metadata)
         metadata.to_json
       end
+
+      def self.print_summary(update_manager)
+        summary = {}
+        update_manager.changes.keys.each do |category|
+          update_category = update_manager.changes[category]
```

Comment:
Can you simplify this to:

```ruby
update_manager.changes.each do |category, update_category|
  update_category = update_category.keys if update_category.respond_to?(:keys)

  ...
end
```

Then I think you could avoid the type checking conditional below.

Reply:
Or if you want to avoid redefining `updaate_category` you could also do something like: `update_files = update_category.respond_to?(:keys) ? update_category.keys : update_category`